packages/providers/test/provider.test.ts

```diff
@@ -177,43 +201,30 @@ describe('Provider', () => {
     expect(nodeVersion).toBeDefined();
   });
 
-  it('can change the provider url of the current instance', () => {
+  it('can change the provider url of the current instance', async () => {
     const providerUrl1 = 'http://127.0.0.1:4000/graphql';
     const providerUrl2 = 'http://127.0.0.1:8080/graphql';
-    const provider = new Provider(providerUrl1);
-    const spyGraphQLClient = jest.spyOn(GraphQL, 'GraphQLClient');
+
+    const provider = new Provider(providerUrl1, {
+      fetch: (url: string, options: FetchRequestOptions) =>
```

Comment:
The mock was giving me issues because `GraphQLClient` internally now receives a custom `fetch` function and the test was failing because of reference issues related to mocking and not because a bug was introduced. This test implementation also verifies that the provider url of the graphql client changed, but does it via the public API and not by assuming how the `GraphQLClient` is constructed.